firefox-ios/firefox-ios-tests/Tests/ClientTests/TabManagement/TabManagerTests.swift

```diff
@@ -1764,15 +1765,15 @@ class TabManagerTests: XCTestCase {
             case .normalActive:
                 tab = Tab(profile: mockProfile, windowUUID: tabWindowUUID)
             case .normalInactive:
-                let lastMonthDate = Date().lastMonth
+                let lastMonthDate = testDate.lastMonth
                 tab = Tab(profile: MockProfile(), windowUUID: tabWindowUUID, tabCreatedTime: lastMonthDate)
             case .privateAny:
                 tab = Tab(profile: mockProfile, isPrivate: true, windowUUID: tabWindowUUID)
             case .normalInactive2Weeks:
-                let twoWeeksDate = Date().lastTwoWeek
+                let twoWeeksDate = testDate.lastTwoWeek
                 tab = Tab(profile: MockProfile(), windowUUID: tabWindowUUID, tabCreatedTime: twoWeeksDate)
             case .normalInactiveYesterday:
-                let yesterdayDate = Date.yesterday
+                let yesterdayDate = yesterdayDate()
```

Comment:
while this does fix the issue for yesterday date, it may be a bit confusing adding this here while we have some other helper methods that we are still using in `Date` extension.

It looks like we were using the current date as `Date.noon` based on `dayBefore` and the others (see screenshot) , could we follow suit + use `noon` as the test date for now? We might have to rethink a better solution, but I think being consistent with other tests and using that versus a specific case here. I do like how we are using `Calendar.current.date(byAdding: components, to: testDate)`  in `yesterdatDate()` instead of `noon`, but might require a bigger refactor. 

Let me know what you think! 

![image](https://github.com/user-attachments/assets/9f6ef715-760c-4562-85af-f1b12653b782)